pallets/pools/src/lib.rs

```diff
@@ -1524,20 +1607,57 @@ pub mod pallet {
 
 					pool.tranches.combine_with_mut_residual_top(
 						tranches.into_iter(),
-						|tranche, (new_tranche_type, seniority)| {
+						|tranche, tranche_update| {
 							// Update debt of the tranche such that the interest is accrued until now with the previous interest rate
 							tranche.accrue(now)?;
 
-							tranche.tranche_type = new_tranche_type.clone();
+							tranche.tranche_type = tranche_update.tranche_type.clone();
 
-							if let Some(new_seniority) = seniority {
+							if let Some(new_seniority) = tranche_update.seniority {
 								tranche.seniority = new_seniority.clone();
 							}
+
 							Ok(())
 						},
 					)?;
 				}
 
+				// Checking if we have a new tranche?
+
+				//
+				// The case when Metadata AND the tranche changed, we don't allow for an or. Both have to be changed (for now)
+				//
+				if let Change::NewValue(metadata) = &changes.tranche_metadata {
+					// I just assume Vec<TrancheMetadata> and Vec<TrancheUpdate> have the same order?
+					if let Change::NewValue(tranches) = &changes.tranches {
+						// It seems like we need to associate metadata with a tranche? I can't seem to find the relation.
+						// The tranche_id in the location? But where does tranche_id come from?
+						for (_, updated_metadata) in tranches.iter().zip(metadata.iter()) {
+							let decimals =
+								T::AssetRegistry::metadata(&pool.currency).unwrap().decimals;
+							let parachain_id = T::ParachainId::get();
+
+							let m = create_asset_metadata(
+								decimals,
+								pool.currency,
+								parachain_id,
+								updated_metadata.clone().token_name.to_vec(),
+								updated_metadata.clone().token_symbol.to_vec(),
+							)?;
+
+							assert_ok!(T::AssetRegistry::update_asset(
+								pool.currency,
+								Some(m.decimals),
+								Some(m.name),
+								Some(m.symbol),
+								Some(m.existential_deposit),
+								Some(m.location),
```

Comment:
It should be the same location, but I am not sure, we should allow to change the location in our domain. WDYT @NunoAlexandre?

Reply:
@gruberb @mustermeiszer yeah, changing the MultiLocation on a tranche token would cause some serious issue with other projects that would integrate with us. This location is essentially the id of the asset in the ecosystem, so if we change it everyone else needs to change it as well as move the balances associated with the old location/internal representation of the asset to the new one.